src/Core/Silk.NET.Core/Contexts/INativeWindow.cs

```diff
@@ -0,0 +1,77 @@
+﻿// This file is part of Silk.NET.
+// 
+// You may modify and distribute Silk.NET under the terms
+// of the MIT license. See the LICENSE file for details.
+
+namespace Silk.NET.Core.Contexts
+{
+    /// <summary>
+    /// Wraps any underlying native window handles a window (or window-like instance) may have. 
+    /// </summary>
+    public interface INativeWindow
+    {
+        /// <summary>
+        /// Gets a bitmask enum representing the underlying platforms powering this window.
+        /// </summary>
+        public NativeWindowFlags Kind { get; }
```

Comment:
IMO types that have a Kind property that basically just indicates what type they really are should just be multiple types.
We have great pattern matching in C# now, so just (empty) INativeWindow and then `INativeWindow[X11/Wayland/WinRT/UIKit/Win32/Vivante/Android/Glfw] : INativeWindow` would be much preferred here.

Reply:
This would introduce much more bloat than required: first there'd have to be lots of different interfaces in Silk.NET.Core, then there'd have to be lots of different types in both GLFW and SDL, and then the windowing backends will have to pick the correct type based on the current context. The last one isn't too much of a worry but the others you'd be exploding the codebase from 3 types to 27 types.

Reply:
🤔 That is a fair point, but not sure if type count is such an important argument.
I'm primarily concerned about how we would annotate this in .net 6, as we can't somehow platform annotate based on a Kind, just the actual type.

Reply:
Sure, but we can annotate the properties as being specific to a certain platform. The kind enum is just another way of being able to tell which platform which properties are "available" on.

Reply:
That's a bit of an odd pattern though, isn't it? `SDL` and `Android` are both possibly available on Android, but also possibly not. So one would have to check whether they are on Android *and* check for both of them being non-null.
Guess this is fair enough for now though, .net 6 will be 3.0 anyways.